src/logger.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from rich.logging import RichHandler
+
+FORMAT = "%(message)s"
+
+class Logger:
+    def __init__(self):
+        self.__FORMAT = "%(message)s"
+        logging.basicConfig(
+                level="NOTSET", format=FORMAT, datefmt="[%X]", handlers=[RichHandler(rich_tracebacks=True)]
+                )
+        self.__logger = logging.getLogger("rich")
+        
+    def stringfy(func):
+        def inner(self, *msg, sep=' ', end='\n'):
+            nmsg = []
+            for v in msg:
+                if not isinstance(v, str):
+                    v = str(v)
+                nmsg.append(v)
+            func(self, *nmsg, sep=sep, end=end)
+        return inner
+                
+    @stringfy
+    def debug(self, *msg, sep=' ', end='\n'):
+        self.__logger.debug(sep.join(msg))
```

Comment:
can you add a comment describing what it does? why @stringify is needed? Why not use self.__logger.debug(f"{msg}") instead?

Reply:
It was done to provide a print like syntax for the logging functions, we can remove it, if it's not needed.